sway-lib-std/src/storage/storage_key.sw

```diff
@@ -98,6 +98,39 @@ impl<T> StorageKey<T> {
         write(self.slot, self.offset, value);
     }
 
+    /// Clears the value at `self`. 
+    ///
+    /// # Number of Storage Accesses
+    ///
+    /// * Clears: `1`
+    ///
+    /// # Examples
+    ///
+    /// ```sway
+    /// fn foo() {
+    ///     let r: StorageKey<u64> = StorageKey {
+    ///         slot: 0x0000000000000000000000000000000000000000000000000000000000000000,
+    ///         offset: 2,
+    ///         field_id: 0x0000000000000000000000000000000000000000000000000000000000000000,
+    ///     };
+    ///     r.write(42);
+    ///
+    ///     let cleared = r.clear();
+    ///     assert(cleared);
+    /// }
+    /// ```
+    #[storage(write)]
+    pub fn clear(self) -> bool {
+        if __size_of::<T>() == 0 {
+            // If the generic doesn't have a size, this is an empty struct and nothing can be stored at the slot.
+            // This clears the length value for StorageVec, StorageString, and StorageBytes 
+            // or any other Storage type.
+            clear::<u64>(self.field_id, 0)
```

Comment:
I'm unsure about the information flow here. We check a generic `T` and then coerce it into a `u64` because of the specific types that have been found to cause an issue.

Over time this might change and there will be a bug here because we assume a `u64`.
I guess if it passes CI now then it's fine and we'll catch it later if it ever becomes an issue.

Reply:
This is done due to a limitation in Rust where function overloading does not exist. Two implementations of `clear()` cannot be done and each case needs to be handled here.
We are unable to check what type the generic passed is. And even if we did call `clear::<T>`, it uses the size of `T` and thus nothing to clear in the first place.